### PR TITLE
Cleanup stuff

### PR DESCRIPTION
**Pull request by wchill**
Non comprehensive list of changes:

* Add pep484 type hinting to all the bot code
* Clean up some of the code paths (also to make type hinting pass)
* pep8 compliance
* Reorganize imports in the following order:
  * `from typing import ...`
  * `from/import` built-in imports
  * `from/import` external imports
  * `from/import` project imports
* Use discord.py `abc.Messageable.send` instead of `Client.send_message` (see [migration notes](https://github.com/Rapptz/discord.py/blob/f84219c2b2f94133ea1d7989c50547042dc7145e/docs/migrating.rst))
* Remove code from top level by wrapping in functions or `if __name__ == '__main__':`

### CODE REVIEW

**Comment by aguynamedjoetoo** [comment]
Will hopefully have some time to give this a look-see tomorrow :D

First glance, btw, `haiku.py` was just jacked from another lib with the link to it addressed. We didn't want to use submodules at the time since it was a single file. So we might just want to cherry pick the changes from there out.

Also there's a few imports/froms that I saw just on my phone that were out of order, style nitpick and all that jazz.

Thanks for the PR!

**Comment by wchill** [comment]
@Joeskyyy this import order is what I use at work, similar to https://github.com/PyCQA/flake8-import-order/blob/master/tests/test_cases/complete_smarkets.py but with the from imports coming before the regular imports. Feel free to change the order if desired, I just feel like regular imports and from imports should be separate groups for readability.

**Comment by wchill** [comment]
Oh yeah, in pep484 all the example code snippets have `typing` above the other stdlib imports, hence I do the same thing.

File: daily_markov.py [hunk 1]
```diff
@@ -45,48 +53,49 @@ def get_history(subreddit, limit):
...
                 if tmp_s is None:
                     break
                 quote_r.append(tmp_s)
-                return '> {}\n\n> ~ {}'.format(unidecode(' '.join(quote_r)), '/r/{}'.format(SUB))
-        else:
-            return ""
+                return '> {}\n\n> ~ {}'.format(unidecode(' '.join(quote_r)), '/r/{}'.format(subreddit.display_name))
+        return ""
+    except IndexError:
+        error_msg = "Error: subreddit '{}' is too dank to simulate.".format(subreddit.display_name)
```
Comment (aguynamedjoetoo) [inline daily_markov.py:86]: LUL

**Review by aguynamedjoetoo — APPROVED** [review]

**Comment by aguynamedjoetoo** [comment]
I'm not against the formatting, works5me.

I do worry a bit about the `haiku.py` changes since it's not our code from the beginning, we're now kind of taking it on as our own. In the unlikely event it bites us, we'll just revert :D

**Comment by wchill** [comment]
the haiku code is licensed MIT so we won't be running into any problems with it, all we need to do is keep the original attribution

**Pull request merged**